File: FrontEnd-ProjectReact/my-react-app-inventory/src/types/itemDetail.ts
import type { ItemImage } from "./itemImage";
import type { Item } from "./item";
import type { DeliveryReceiptDetail } from "./deliveryReceiptDetail";
import type { ItemInventory } from "./itemInventory";
import type { ItemSupplier } from "./itemSupplier";
import type { ItemUnitMeasure } from "./itemUnitMeasure";
import type { PurchaseRequisitionDetail } from "./purchaseRequisitionDetail";
import type { PurchaseOrderDetail } from "./purchaseOrderDetail";
import type { PurchaseReturnDetail } from "./purchaseReturnDetail";
import type { ReceivingReportDetail } from "./receivingReportDetail";
import type { SalesInvoiceDetail } from "./salesInvoiceDetail";
import type { SalesOrderDetail } from "./salesOrderDetail";
import type { SalesReturnDetail } from "./salesReturnDetail";
import type { StockCountDetail } from "./stockCountDetail";


export interface ItemDetail {
  id: number;
  ItemDetailCode: string;
  ItemId: string; // Note: If it's an int in C#, change to number
  ItemDetailNo: string;
  Barcode?: string;
  SerialNo?: string;
  PartNo?: string;
  WarehouseCode: string;
  LocationCode: string;
  Volume?: string;
  Size?: string;
  Weight?: string;
  ExpiryDate?: string;
  UnitMeasure?: string;
  Unitprice?: number;
  Warranty?: string;
  ModifiedDateTime: string;
  CreatedDateTime: string;
  Eoq?: number;
  Rop?: number;
  RecStatus?: string;

  Item: Item | null;
  ItemImage: ItemImage | null;
  DeliveryReceiptDetailFile?: DeliveryReceiptDetail[] | null;
  ItemInventory?: ItemInventory[] | null;
  ItemSupplier?: ItemSupplier[] | null;
  ItemUnitMeasure?: ItemUnitMeasure[] | null;
  PurchaseOrderDetail?: PurchaseOrderDetail[] | null;
  PurchaseRequisitionDetail?: PurchaseRequisitionDetail[] | null;
  PurchaseReturnDetail?: PurchaseReturnDetail[] | null;
  ReceivingReportDetail?: ReceivingReportDetail[] | null;
  SalesInvoiceDetail?: SalesInvoiceDetail[] | null;
  SalesOrderDetail?: SalesOrderDetail[] | null;
  SalesReturnDetail?: SalesReturnDetail[] | null;
  StockCountDetail?: StockCountDetail[] | null;
}

File: Sample/Sample.Domain/Entities/ItemDetail.cs
using System;
using System.Collections.Generic;
using Sample.Domain.Domain;
using Sample.Domain.Entities;

namespace Sample.Domain.Entities
{
    public class ItemDetail : EntityBase, IAggregateRoot
    {
        public virtual int id { get; set; }
        public virtual string ItemDetailCode { get; set; } = null!;
        public virtual string ItemId { get; set; } = null!;
        public virtual string ItemDetailNo { get; set; } = null!;
        public virtual string? Barcode { get; set; }
        public virtual string? SerialNo { get; set; }
        public virtual string? PartNo { get; set; }
        public virtual string WarehouseCode { get; set; } = null!;
        public virtual string LocationCode { get; set; } = null!;
        public virtual string? Volume { get; set; }
        public virtual string? Size { get; set; }
        public virtual string? Weight { get; set; }
        public virtual string? ExpiryDate { get; set; }
        public virtual string? UnitMeasure { get; set; }
        public virtual decimal? Unitprice { get; set; }
        public virtual string? Warranty { get; set; }
        public virtual DateTime ModifiedDateTime { get; set; }
        public virtual DateTime CreatedDateTime { get; set; }
        public virtual int? Eoq { get; set; }
        public virtual int? Rop { get; set; }
        public virtual string? RecStatus { get; set; }


        public virtual Item? ItemMaster { get; set; }
        public virtual ItemImage? ItemImage { get; set; }
        public virtual ICollection<DeliveryReceiptDetail>? DeliveryReceiptDetailFile { get; set; }
        //public virtual ICollection<InventoryAdjustment> InventoryAdjustment { get; set; }
        //public virtual ICollection<InventoryTransfer> InventoryTransfer { get; set; }
        public virtual ICollection<ItemInventory>? ItemInventory { get; set; }
        public virtual ICollection<ItemSupplier>? ItemSupplier { get; set; }
        public virtual ICollection<ItemUnitMeasure>? ItemUnitMeasure { get; set; }
        public virtual ICollection<PurchaseOrderDetail>? PurchaseOrderDetail { get; set; }
        public virtual ICollection<PurchaseRequisitionDetail>? PurchaseRequisitionDetail { get; set; }
        public virtual ICollection<PurchaseReturnDetail>? PurchaseReturnDetail { get; set; }
        public virtual ICollection<ReceivingReportDetail>? ReceivingReportDetail { get; set; }
        public virtual ICollection<SalesInvoiceDetail>? SalesInvoiceDetail { get; set; }
        public virtual ICollection<SalesOrderDetail>? SalesOrderDetail { get; set; }
        public virtual ICollection<SalesReturnDetail>? SalesReturnDetail { get; set; }
        //public virtual ICollection<StockCountDetail> StockCountDetail { get; set; }

        public static ItemDetail Create(ItemDetail itemDetail, string createdBy)
        {
            //Place your Business logic here
            itemDetail.Created_by = createdBy;
            itemDetail.Date_created = DateTime.Now;
            return itemDetail;
        }

        public static ItemDetail Update(ItemDetail itemDetail, string editedBy)
        {
            //Place your Business logic here
            itemDetail.Edited_by = editedBy;
            itemDetail.Date_edited = DateTime.Now;
            return itemDetail;
        }
    }
}
